algorithms/unit_tests/TestStdAlgorithmsConstraints.cpp

```diff
@@ -81,5 +81,122 @@ TEST(std_algorithms, is_admissible_to_std_algorithms) {
                strided_view_3d_t>::value);
 }
 
+TEST(std_algorithms, expect_no_overlap) {
+  namespace KE     = Kokkos::Experimental;
+  using value_type = double;
+
+  static constexpr size_t extent0 = 13;
+
+  //-------------
+  // 1d views
+  //-------------
+  using static_view_1d_t = Kokkos::View<value_type[extent0]>;
+  [[maybe_unused]] static_view_1d_t static_view_1d{
+      "std-algo-test-1d-contiguous-view-static"};
+
+  using dyn_view_1d_t = Kokkos::View<value_type*>;
+  [[maybe_unused]] dyn_view_1d_t dynamic_view_1d{
+      "std-algo-test-1d-contiguous-view-dynamic", extent0};
+
+  using strided_view_1d_t = Kokkos::View<value_type*, Kokkos::LayoutStride>;
+  Kokkos::LayoutStride layout1d{extent0, 2};
+  [[maybe_unused]] strided_view_1d_t strided_view_1d{
+      "std-algo-test-1d-strided-view", layout1d};
+
+#if !defined(NDEBUG) || defined(KOKKOS_ENFORCE_CONTRACTS) || \
+    defined(KOKKOS_ENABLE_DEBUG)
```

Comment:
I suspect that guard is not right.
You guarded the precondition check with `#ifdef KOKKOS_ENABLE_DEBUG` so it probably should be something like
```suggestion
#if defined(KOKKOS_ENABLE_DEBUG) && !defined(NDEBUG)
```